src/main/java/randoop/contract/ObserverEqArray.java

```diff
@@ -0,0 +1,217 @@
+      return true;
+    }
+    return PrimitiveTypes.isBoxedPrimitive(cls) || cls.isPrimitive();
+  }
+
+  @Override
+  public String toCodeString() {
+    StringBuilder b = new StringBuilder();
+    if (Array.getLength(value) < MAX_ARRAY_LENGTH) {
+      if (value.getClass().getComponentType() == float.class
+          || value.getClass().getComponentType() == double.class) {
+        b.append(
+            String.format(
+                "org.junit.Assert.assertArrayEquals(x0, %s, %s);",
+                printArray(), FLOATING_POINT_DELTA));
+      } else {
+        b.append(String.format("org.junit.Assert.assertArrayEquals(x0, %s);", printArray()));
+      }
+    }
+    return b.toString();
+  }
+
+  /**
+   * Prints the code string of the second parameter (instantiation of array) in assertArrayEquals
+   * e.g. prints the second parameter in assertArrayEquals(var, new int[] {1,2,3})
+   *
+   * @return String that represents an instantiation of an array equal to value
+   */
+  private String printArray() {
+    String finalString = "";
+    if (value == null) {
+      finalString += "null";
+    } else {
+      finalString += "new ";
+      finalString += value.getClass().getCanonicalName();
+      finalString += printArrayComponents();
+    }
+    return finalString;
+  }
+
+  /**
+   * Converts an Object into its array representation
+   *
+   * @param value - the given Object
+   * @return its array representation
+   */
+  private Object[] toObjectArray(Object value) {
+    if (value instanceof byte[]) {
+      return ArrayUtils.toObject((byte[]) value);
+    } else if (value instanceof short[]) {
+      return ArrayUtils.toObject((short[]) value);
+    } else if (value instanceof int[]) {
+      return ArrayUtils.toObject((int[]) value);
+    } else if (value instanceof long[]) {
+      return ArrayUtils.toObject((long[]) value);
+    } else if (value instanceof float[]) {
+      return ArrayUtils.toObject((float[]) value);
+    } else if (value instanceof double[]) {
+      return ArrayUtils.toObject((double[]) value);
+    } else if (value instanceof char[]) {
+      return ArrayUtils.toObject((char[]) value);
+    } else if (value instanceof boolean[]) {
+      return ArrayUtils.toObject((boolean[]) value);
+    } else {
+      return (Object[]) value;
+    }
+  }
+
+  /**
+   * Helper method that prints the components of the array
+   *
+   * @return String that represents the components of the array
+   */
+  private String printArrayComponents() {
+    String finalString = "{";
+    int length = Array.getLength(value);
+    if (value.getClass().getComponentType().isEnum()) {
+      for (int i = 0; i < length; i++) {
+        if (Array.get(value, i) == null) {
+          finalString += "null";
+        } else {
+          finalString += new EnumValue((Enum<?>) Array.get(value, i)).getValueName();
+        }
+        if (i < length - 1) {
+          finalString += ", ";
```

Comment:
You don't need this logic.  Use StringJoiner instead.  That is also more efficient than repeated String concatenation.